### PR TITLE
feat(node): add events support

### PR DESCRIPTION
**Pull request by kt3k**
This PR implements DOM Event handlings using Deno's EventTarget implementation.

### CODE REVIEW

File: src/dom/node.ts [hunk 1]
```diff
@@ -443,6 +429,16 @@ export class Node extends EventTarget {
     // intended logic
     return Node.DOCUMENT_POSITION_FOLLOWING;
   }
+
+  getRootNode(opts: { composed?: boolean } = {}): Node {
+    if (this.parentNode) {
+      return this.parentNode.getRootNode(opts);
+    }
+    if (opts.composed && (this as any).host) {
+      (this as any).host.getRootNode(opts);
```
Comment (b-fuze) [inline src/dom/node.ts:438]: Shouldn't this have a return statement?

File: test/units/Node-events.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+Deno.test("Node.dispatchEvent, event bubbles", async () => {
+  const doc = new DOMParser().parseFromString(
+    `<div class="parent"><div class="child"></div></div>`,
+    "text/html",
+  )!;
+  const parent = doc.querySelector(".parent")!;
+  const child = doc.querySelector(".child")!;
+  let called = false;
+  // FIXME(kt3k): this empty event handler is necessary to work around
+  // the bug in EventTarget impl in Deno.
```
Comment (b-fuze) [inline test/units/Node-events.ts:30]: Is there an issue open for this? Maybe we can include it in the comment here

Comment (kt3k) [reply test/units/Node-events.ts:30]: No issue yet, but this line causes the problem https://github.com/denoland/deno/blob/07618c861e27e284cae3a309783ef78dcef611dd/ext/web/02_event.js#L976-L979

It only checks the dispatched element's handler types no matter its ancestors have handlers of the given type

File: src/dom/node.ts (hunk 1 above)
Comment (kt3k) [reply src/dom/node.ts:438]: Good catch! Thanks!

**Comment by b-fuze** [comment]
LGTM. Thanks

**Comment by kt3k** [comment]
Thanks for landing!

**Pull request merged**